src/main/java/org/arquillian/container/chameleon/ChameleonConfiguration.java

```diff
@@ -141,4 +141,18 @@ private String getOutputDirectory() {
             }
         }
     }
+
+    public static boolean isSupported(String source) throws Exception {
+        Target target = Target.from(source);
+        ChameleonConfiguration chameleonConfiguration = new ChameleonConfiguration();
+        Container[] containers = new ContainerLoader().load(chameleonConfiguration.getChameleonContainerConfigurationFileStream(), chameleonConfiguration.getChameleonResolveCacheFolder());
+        for (Container container : containers) {
```

Comment:
Chameleon is Java8 project, so you can use lambdas if you want to simplify all for logic.